### PR TITLE
Added `feature_thread`

### PR DESCRIPTION
**Pull request by eleensmathew**
Most of the information is from https://github.com/matrix-org/matrix-doc/blob/gsouquet/threading-via-relations/proposals/3440-threading-via-relations.md. Please review and tell me if any changes are to be made.

---
This PR currently has no changelog labels, so will not be included in changelogs.

A reviewer can add one of: `T-Deprecation`, `T-Enhancement`, `T-Defect`, `T-Task` to indicate what type of change this is, or add `Type: [enhancement/defect/task]` to the description and I'll add them for you.

### CODE REVIEW

**Review by gsouquet — CHANGES_REQUESTED** [review]
Thank you for opening a PR here.
The idea is to document the feature at a very high level so that folks understand roughly what it does.

For this pull request to be merged I would recommend you remove all technical details and write a few lines describing what this does

File: docs/labs.md [hunk 1]
```diff
@@ -172,3 +172,14 @@ who prefer to have the right panel open consistently across rooms.
...
 used in that room is shown.
+
+## Threaded Messaging (`feature_thread`)
+
+Threading allows users to branch out a new conversation from the main timeline of a room. This is particularly useful in high traffic rooms where multiple conversations can happen in parallel or when a single discussion might stretch over a very long period of time.
+
+When a thread head is aggregated (as in MSC2675), it returns a summary of the thread: the latest message, a list of participants and the total count of messages. I.e. in places which include bundled relations (per MSC2675).
+
+A thread will be displayed as a chain of replies on clients unaware of threads.
+Thread-ready clients would recieve the latest event in the thread unless a user is explicitly replying to another event.
```
[the remark this answers is not in the document]
Comment (eleensmathew) [reply docs/labs.md:183]: ## Threaded Messaging (`feature_thread`)

Threading allows users to branch out a new conversation from the main timeline of a room. This is particularly useful in high traffic rooms where multiple conversations can happen in parallel or when a single discussion might stretch over a very long period of time.

Clients should fill in the new field (https://github.com/matrix-org/matrix-doc/blob/gsouquet/threading-via-relations/proposals/3440-threading-via-relations.md#backwards-compatibility) in order to display that in a thread context.
When a thread head is aggregated (as in MSC2675), it returns a summary of the thread: the latest message, a list of participants and the total count of messages.
A thread will be displayed as a chain of replies on clients unaware of threads.
Thread-ready clients would receive the latest event in the thread unless a user is explicitly replying to another event.

Clients can also synthesize read receipts but it is possible that some notifications get lost on a fresh start.

[the remark this answers is not in the document]
Comment (eleensmathew) [reply docs/labs.md:183]: >Is this okay?? should I remove the link and not include the  'fill the feild' part? Also should i remove the (As in MSC2675) part as well?

[the remark this answers is not in the document]
Comment (rriemann) [reply docs/labs.md:183]: Is it OK @gsouquet ?

File: docs/labs.md [hunk 2]
```diff
@@ -172,3 +172,14 @@ who prefer to have the right panel open consistently across rooms.
...
+
+Clients can synthesize read receipts but it is possible that some notifications get lost on a fresh start.
```
Comment (gsouquet) [inline docs/labs.md:185]:
```suggestion
Threading allows users to branch out a new conversation from the main timeline of a room. This is particularly useful in high traffic rooms where multiple conversations can happen in parallel or when a single discussion might stretch over a very long period of time.

Threads can be access by clicking their summary below the root event on the room timeline. Users can find a comprehensive list of threads by click the icon on the room header button.

This feature might work in degraded mode if the homeserver a user is connected to does not advertise support for the unstable feature `org.matrix.msc3440`  when calling the `/versions` API endpoint.
```

**Review by gsouquet — CHANGES_REQUESTED** [review]

**Comment by gsouquet** [comment]
It looks like this PR needs rebasing too, there's some conflicting file

**Comment by eleensmathew** [comment]
Hi @gsouquet  I changed the file as you suggested, but I opened a new pull request https://github.com/vector-im/element-web/pull/21303 to fix the issues.

**Comment by gsouquet** [comment]
Closing in favour of https://github.com/vector-im/element-web/pull/21303

**Pull request closed without merging**